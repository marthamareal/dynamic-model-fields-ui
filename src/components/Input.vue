<template>
<form class="form-group" ref="form" @submit.stop.prevent="handleSubmit">
  <div role="group">
  <label for="input-live"><strong>{{name}}</strong></label>
  <!-- Check if typoe is select or not -->
    <div v-if="field.field_type === 'select-one'">
      <b-form-group v-for="(option, index) in field.options" :key="index" class="form-check form-check-inline">
        <b-form-radio name="some-radios" value="A">{{option}}</b-form-radio>
      </b-form-group>
    </div>
    <div v-else>
      <b-form-input
        :name="field.name"
        :id="modalId"
        :type="field.field_type"
        trim
      ></b-form-input>
   </div>
   <!-- Check ends -->
  </div>
</form>
</template>

<script>
import BFormInput from 'bootstrap-vue/es/components/form-input/form-input'

export default {
  name: 'Input',
  props: {
    field: Object,
    modalId: String
  },
  components: {
    'b-form-input': BFormInput
  },
  data (props) {
    return { name: props.field.name.charAt(0).toUpperCase() + props.field.name.slice(1) }
  }
}

</script>
