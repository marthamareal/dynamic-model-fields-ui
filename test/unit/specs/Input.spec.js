import { mount } from '@vue/test-utils'

import Input from '@/components/Input.vue'

describe('Input.vue', () => {
    const wrapper = mount(Input, {
        propsData: {
            field: {
                "name": "date",
                "field_type": "date",
                "options": null
            }
          }
    })
    it('should render as expected', () => {
        expect(wrapper.html()).toMatchSnapshot();
})
})