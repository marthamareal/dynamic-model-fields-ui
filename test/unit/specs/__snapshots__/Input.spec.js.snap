// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input.vue should render as expected 1`] = `
<form class="form-group">
    <div role="group"><label for="input-live"><strong>Date</strong></label>
        <div>
            <b-form-input id="input-live" type="date" trim=""></b-form-input>
        </div>
    </div>
</form>
`;
